server/src/test/java/org/elasticsearch/action/bulk/FailureStoreDocumentConverterTests.java

```diff
@@ -24,13 +28,16 @@
 
 public class FailureStoreDocumentConverterTests extends ESTestCase {
 
-    public void testFailureStoreDocumentConverstion() throws Exception {
+    public void testFailureStoreDocumentConversion() throws Exception {
         IndexRequest source = new IndexRequest("original_index").routing("fake_routing")
             .id("1")
             .source(JsonXContent.contentBuilder().startObject().field("key", "value").endObject());
 
         // The exception will be wrapped for the test to make sure the converter correctly unwraps it
-        Exception exception = new ElasticsearchException("Test exception please ignore");
+        ElasticsearchException exception = new ElasticsearchException("Test exception please ignore");
+        exception.addHeader(CompoundProcessor.PIPELINE_ORIGIN_EXCEPTION_HEADER, Arrays.asList("some-failing-pipeline", "some-pipeline"));
+        exception.addHeader(CompoundProcessor.PROCESSOR_TAG_EXCEPTION_HEADER, "foo-tag");
+        exception.addHeader(CompoundProcessor.PROCESSOR_TYPE_EXCEPTION_HEADER, "bar-type");
         exception = new RemoteTransportException("Test exception wrapper, please ignore", exception);
```

Comment:
We could use `randomBoolean()` to randomly wrap the exception, that way we cover both wrapped and unwrapped cases. Might be worth adding a random chance of the exception is suppressed as well?

Reply:
Randomizing the wrapping sounds like a good idea to me. But I'm not sure suppressing would work, as [this](https://github.com/nielsbauman/elasticsearch/blob/425ff27af0cb9e413aee6ac0633757a0cdcf9164/server/src/main/java/org/elasticsearch/action/bulk/FailureStoreDocumentConverter.java#L86) line doesn't take into account suppressed exceptions. [This](https://github.com/nielsbauman/elasticsearch/blob/425ff27af0cb9e413aee6ac0633757a0cdcf9164/server/src/main/java/org/elasticsearch/action/bulk/FailureStoreDocumentConverter.java#L115) line does take suppressed exceptions into account. I guess that means that it could be possible that the `(type|message|stack_trace)` and `(pipeline|processor)_*` fields can be referring to different exceptions? Is that something we want to align?

Reply:
I don't think it needs to be aligned per se. I would trust whoever is doing the suppressing that the new exception is more important to return to the user than the suppressed ingest exception, but that ingest exception still occurred as part of this failure. We should capture the data off of it still.